BasicParam:
  control_method: 2 # 2-NMPC  3-LMPC
  data_record: 1
  land_flag: 0
  delay_flag: 0

NmpcParam:
  Np: 20
  Nc: 20

  w_P_xy: 500 
  w_P_z: 500
  w_V_xy: 10
  w_V_z: 0
  w_q: 80

  w_P_xy_n: 500
  w_P_z_n: 500
  w_V_xy_n: 0
  w_V_z_n: 0
  w_q_n: 40
  
  w_T: 3
  w_omega_x: 15
  w_omega_y: 15
  w_omega_z: 15

LmpcParam:
  Np: 20
  Nc: 20

  w_P_xy: 100 
  w_P_z: 100
  w_V_xy: 5
  w_V_z: 1

  w_P_xy_n: 200
  w_P_z_n: 50
  w_V_xy_n: 10
  w_V_z_n: 2

  w_U_x: 2
  w_U_y: 2
  w_U_z: 1

UAVParam:
  mass: 1.52
  arm_length: 0.225
  hover_throttle: 0.466 # 0.47
  max_tilt: 0.349066 # pi/9
  max_accel_xy: 3.0
  max_accel_z: 1.0

ControlParam:
  target_z: 0.8
  land_time1: 100
  land_time2: 30

  limit_vel: 2.0
  k_vel_x: 6.0
  k_vel_y: 6.0
  k_vel_z: 3.6
  y_kp: 0.7
  y_ki: 0.01
  y_kd: 0.2
  yaw_kp: 2
  yaw_ki: 0.05
  yaw_kd: 0.2
